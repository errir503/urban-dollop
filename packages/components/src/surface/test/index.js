/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Surface } from '../index';

describe( 'props', () => {
	let base;
	beforeEach( () => {
		base = render( <Surface>Surface</Surface> );
	} );

	test( 'should render correctly', () => {
		expect( base.container.firstChild ).toMatchSnapshot();
	} );

	test( 'should render variants', () => {
		const { container } = render(
			<Surface variant="secondary">Surface</Surface>
		);
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );

	test( 'should render borderLeft', () => {
		const { container } = render( <Surface borderLeft>Surface</Surface> );
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );

	test( 'should render borderRight', () => {
		const { container } = render( <Surface borderRight>Surface</Surface> );
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );

	test( 'should render borderTop', () => {
		const { container } = render( <Surface borderTop>Surface</Surface> );
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );

	test( 'should render borderBottom', () => {
		const { container } = render( <Surface borderBottom>Surface</Surface> );
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );
} );
