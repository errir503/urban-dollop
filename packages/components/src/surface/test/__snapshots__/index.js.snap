// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render borderBottom 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-surface css-k1ws5-View-Surface-getBorders-primary em57xhy0"
+   class="components-surface css-avigp2-View-Surface-getBorders-primary em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
`;

exports[`props should render borderLeft 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-surface css-gi9sau-View-Surface-getBorders-primary em57xhy0"
+   class="components-surface css-avigp2-View-Surface-getBorders-primary em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
`;

exports[`props should render borderRight 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-surface css-1lwskr8-View-Surface-getBorders-primary em57xhy0"
+   class="components-surface css-avigp2-View-Surface-getBorders-primary em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
`;

exports[`props should render borderTop 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-surface css-sdy6q1-View-Surface-getBorders-primary em57xhy0"
+   class="components-surface css-avigp2-View-Surface-getBorders-primary em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  color: #000;
  position: relative;
}

<div
  class="components-surface emotion-0 emotion-1"
  data-wp-c16t="true"
  data-wp-component="Surface"
>
  Surface
</div>
`;

exports[`props should render variants 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-surface css-jkj9zl-View-Surface-getBorders-secondary em57xhy0"
+   class="components-surface css-avigp2-View-Surface-getBorders-primary em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Surface"
  >
    Surface
  </div>
`;
